src/state/actions/dispatchArIOInteraction.ts

```diff
@@ -202,60 +199,42 @@ export default async function dispatchArIOInteraction({
         }
         break;
       case ARNS_INTERACTION_TYPES.PRIMARY_NAME_REQUEST: {
+        console.log('herherhehr', arioContract.process.processId);
+        result = await arioContract.setPrimaryName(
+          {
+            name: payload.name,
+          },
+          {
+            ...WRITE_OPTIONS,
+            onSigningProgress: (step, payload) => {
+              if (step === 'requesting-primary-name') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Requesting primary name '${payload.name}'`,
+                });
+              } else if (step === 'request-already-exists') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Primary name request for '${payload.name}' already exists!`,
+                });
+              } else if (step === 'approving-request') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Approving primary name request for '${payload.name}'`,
+                });
+              }
+            },
+          },
+        );
+        // send a final confirmation message
```

Comment:
cranking could lead to this take a few minutes to apply, it's likely a good idea to notify the user of that in this modal and show it for a few seconds at least before closing.